platform/kube/cache.go

```diff
@@ -53,6 +59,9 @@ func newPodCache(ch cacheHandler) *PodCache {
 
 // getPodByIp returns the pod or nil if pod not found or an error occurred
 func (pc *PodCache) getPodByIP(addr string) (*v1.Pod, bool) {
+	pc.rwMu.RLock()
+	defer pc.rwMu.RUnlock()
```

Comment:
This looks fine.
Can you ensure that same problem is not present in getByKey ?

Also tests?

Reply:
Looking at GetByKey, tests coming in a separate PR